test/qa-tests/jstests/dump/no_sharded_secondary_reads.js

```diff
@@ -51,23 +54,26 @@
   runMongoProgram("mongodump", "--host", st.s.host, "-vvvv");
   assert.eq(replDB.system.profile.find(profQuery).count(), 4, "queries are routed to primary");
   printjson(replDB.system.profile.find(profQuery).toArray());
-  assert.eq(replDB.system.profile.find({
-    ns: "test.a",
-    op: "query",
-    $or: [
-      // 4.0
-      {"command.hint._id": 1},
+  // in a mmapv1 stored database, we should snapshot or have a query hint set.
+  if (isMMAPV1) {
+    assert.eq(replDB.system.profile.find({
+      ns: "test.a",
+      op: "query",
+      $or: [
+        // 4.0
+        {"command.hint._id": 1},
 
-      // 3.6 schema
-      {"command.$snapshot": true},
-      {"command.snapshot": true},
+        // 3.6 schema
+        {"command.$snapshot": true},
+        {"command.snapshot": true},
 
-      // 3.4 and previous schema
-      {"query.$snapshot": true},
-      {"query.snapshot": true},
-      {"query.hint._id": 1},
-    ]
-  }).count(), 1);
+        // 3.4 and previous schema
+        {"query.$snapshot": true},
+        {"query.snapshot": true},
+        {"query.hint._id": 1},
+      ]
+    }).count(), 1);
+  }
```

Comment:
Can we check in the else case that we didn't see any snapshot/hints?  That will confirm that we do the right thing for full database dumps.